### ISSUE TITLE
What does the training loss curve look like?

### ISSUE THREAD

**Issue by scurkovic** [AnswerDotAI/ModernBERT#232]
Hi. I am pre-training an encoder-decoder model from scratch that has the same size and almost identical architecture to ModernBERT-base. I used identical hyperparameters (lr warmup, bs warmup, etc.). The loss curve looks weird however, as you can see here:

![Image](https://github.com/user-attachments/assets/ec653f83-f868-433c-8ee6-c1c4f22b0b2e)

Since, this is a different model, obviously, I am only curious if you've observed similar behavior with any of your training runs? Feel free to close this immediately. Thank you.

**Comment by timpal0l** [comment]
Did you shuffle your dataset?

**Comment by scurkovic** [comment]
@timpal0l In that run I didn't as I assumed fineweb was in a random order anyway. In any case, it did change a few things (including adding shuffling, going from UL2 to T5 pretraining task, and changing the optimizer) and it trains fine now.
